src/test/custom_node_buffersqueue_test.cpp

```diff
@@ -0,0 +1,136 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <cstring>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../custom_nodes/common/buffersqueue.hpp"
+
+using namespace ovms;
+using custom_nodes_common::BuffersQueue;
+
+TEST(CustomNodeBuffersQueue, GetAllBuffers) {
+    const std::string content{"abc"};
+    size_t buffersCount = 12;
+    BuffersQueue buffersQueue(content.size(), buffersCount);
+    std::vector<void*> buffers(buffersCount);
+    for (size_t i = 0; i < buffersCount; ++i) {
+        buffers[i] = buffersQueue.getBuffer();
+        ASSERT_NE(nullptr, buffers[i]) << "Failed to get: " << i;
+        std::memcpy(buffers[i], content.c_str(), content.size());
+    }
+    for (size_t i = 0; i < buffersCount; ++i) {
+        EXPECT_EQ(0, std::memcmp(buffers[i], content.c_str(), content.size()))
+            << "Buffer: " << i << " has different content";
+    }
+    std::sort(buffers.begin(), buffers.end());
+    char* ptr = static_cast<char*>(buffers[0]);
+    for (size_t i = 1; i < buffersCount; ++i) {
+        EXPECT_GE(static_cast<char*>(buffers[i]) - ptr, content.size())
+            << "distance between two buffers is to small. Between: " << i - 1 << " and " << i;
+        ptr = static_cast<char*>(buffers[i]);
+    }
+    for (size_t i = 0; i < buffersCount; ++i) {
+        EXPECT_TRUE(buffersQueue.returnBuffer(buffers[i])) << "failed to release buffer: " << i;
+    }
+}
+
+void getBufferAndReturn(BuffersQueue& buffersQueue) {
+    void* buffer = buffersQueue.getBuffer();
+    ASSERT_NE(nullptr, buffer) << "Failed to get buffer";
+}
+
+TEST(CustomNodeBuffersQueue, GetAllBuffersThenNullptrForNextRequest) {
+    const std::string content{"abc"};
+    size_t buffersCount = 1;
+    BuffersQueue buffersQueue(content.size(), buffersCount);
+    std::vector<void*> buffers(buffersCount);
+    for (size_t i = 0; i < buffersCount; ++i) {
+        buffers[i] = buffersQueue.getBuffer();
+        ASSERT_NE(nullptr, buffers[i]) << "Failed to get: " << i;
+    }
+    void* buffer = buffersQueue.getBuffer();
+    EXPECT_EQ(nullptr, buffer) << "Failed to get buffer";
```

Comment:
Maybe we could check in this test one more thing: `getBuffer` after returning buffers does not return `nullptr`

Reply:
GetAndReturnBuffersSeveralTimes this test does it several times